### PR TITLE
icds postgresql performance tuning, fogbugz 248996

### PR DESCRIPTION
**Pull request by javierwilson**
adding performance tuning options to ICDS, and postgresql.conf template support for these options. @snopoke @kaapstorm

### CODE REVIEW

File: ansible/vars/icds/icds_public.yml [hunk 1]
```diff
@@ -61,6 +61,9 @@ postgres_users: "{{ secrets.POSTGRES_USERS }}"
 
 postgres_config:
   max_connections: 300
+  work_mem: '8MB'
+  shared_buffers: '512MB'
```
Comment (benrudolph) [inline ansible/vars/icds/icds_public.yml]: >  a reasonable starting value for shared_buffers is 25% of the memory in your system

how come we aren't using that heuristic?

File: ansible/vars/icds/icds_public.yml [hunk 2]
```diff
@@ -61,6 +61,9 @@ postgres_users: "{{ secrets.POSTGRES_USERS }}"
...
+  effective_cache_size: '8GB'
```
Comment (benrudolph) [inline ansible/vars/icds/icds_public.yml]: > Setting effective_cache_size to 1/2 of total memory would be a normal conservative setting, and 3/4 of memory is a more aggressive but still reasonable amount.

also for this one, what is the ram on our db machine?

**Comment by snopoke** [comment]
@benrudolph good point re memory numbers, one factor to consider is that we don't have dedicate PG machines, they are also running other processes e.g. Riak

File: ansible/vars/icds/icds_public.yml (hunk 2 above)
Comment (javierwilson) [inline ansible/vars/icds/icds_public.yml]: Same as shared_buffers, I wanted to increase and test.  Until we reached 25% and 1/2, just being cautious.

Comment (benrudolph) [inline ansible/vars/icds/icds_public.yml]: ok sounds good 👍

File: ansible/vars/icds/icds_public.yml (hunk 1 above)
Comment (javierwilson) [inline ansible/vars/icds/icds_public.yml]: We increased from 128MB to 512MB, I want to increase and test a few times before we go as high as recommended, we can try 2GB next and 8GB... but not if the system is not actually taking advantage of it.

**Pull request merged**